2018/1/solution.py

```diff
@@ -0,0 +1,25 @@
+import sys
+
+def frequency_total(frequencies):
+    return sum(frequencies)
+
+def frequency_duplicate(frequencies):
+    previous_frequencies = []
+    for frequency in _generate_frequencies(frequencies):
+        if frequency in previous_frequencies:
+            return frequency
+        previous_frequencies.append(frequency)
+
+def _generate_frequencies(frequencies):
+    frequency = 0
+    index = 0
+    while True:
+        yield frequency
```

Comment:
I like the use of generators.  That's a good way of handling infinite sequences.